### PR TITLE
4525 add new what is your immigration status page

### PR DESCRIPTION
**Pull request by avinhurry**
## Context

Amendments to the right to work flow.

Feature flag and redundant logic cleanup will be done in a follow up PR.

## Changes proposed in this pull request

- Surface immigration status page and remove the expandable text box from the original page
- Remove immigration route views, controller, routes and form

- Use `right_to_work_or_study` column instead
- Use `right_to_work_or_study_details` column instead 

[Use right_to_work_or_study_details column instead](https://github.com/DFE-Digital/apply-for-teacher-training/pull/6657/commits/32b5cdbd0745299f4a8ba74f76323532e9881495)

## Guidance to review

- Do we need a RC check?

- Edit `Nationality` and select `Citizen of a different country`
- Choose `Yes` to the next question.
- Notice how the text field no longer expands, but instead you are taken to a new page.
- Play around, look for bugs, leave comments.

## Link to Trello card
https://trello.com/c/9qNLUUww/4525-add-new-what-is-your-immigration-status-page
https://trello.com/c/oUWLYqAJ/4499-remove-question-from-right-to-work-flow

### CODE REVIEW

**Review by frankieroberto — CHANGES_REQUESTED** [review]
Minor bug with the error state for "What is your immigration status?"

Steps to reproduce:

1. Select 'French' as a nationality
2. Answer 'Yes' to 'Do you already have the right to work or study in the UK?'
3. Press 'Save and continue' without selecting an option

Issues:

1. Error message is "can't be blank" - should be "Select your immigration status" or similar.
2. The radios should have a red border to the left of all them, and should repeat the error message beneath the question / h1
3. The link in the error summary should focus the first radio option
4. "Other" is selected - no radios should be selected.

Screenshot: 
 
Screenshot 2022-03-11 at 10 35 05

File: app/forms/candidate_interface/immigration_right_to_work_form.rb [hunk 1]
```diff
@@ -2,43 +2,34 @@ module CandidateInterface
   class ImmigrationRightToWorkForm
     include ActiveModel::Model
 
-    attr_accessor :immigration_right_to_work
+    attr_accessor :right_to_work_or_study, :right_to_work_or_study_details
```
Comment (frankieroberto) [inline app/forms/candidate_interface/immigration_right_to_work_form.rb:5]: The second attribute ('details') could be dropped now, as that's now on a separate form?

```suggestion
    attr_accessor :right_to_work_or_study
```

Comment (avinhurry) [reply app/forms/candidate_interface/immigration_right_to_work_form.rb]: Good spot, have updated ♻️

**Comment by avinhurry** [comment]
> Minor bug with the error state for "What is your immigration status?"
> 
> Steps to reproduce:
> 
>     1. Select 'French' as a nationality
> 
>     2. Answer 'Yes' to 'Do you already have the right to work or study in the UK?'
> 
>     3. Press 'Save and continue' without selecting an option
> 
> 
> Issues:
> 
>     1. Error message is "can't be blank" - should be "Select your immigration status" or similar.
> 
>     2. The radios should have a red border to the left of all them, and should repeat the error message beneath the question / h1
> 
>     3. The link in the error summary should focus the first radio option
> 
>     4. "Other" is selected - no radios should be selected.
> 
> 
> Screenshot:
> 
> Screenshot 2022-03-11 at 10 35 05

Thanks @frankieroberto, have fixed that now. Ready for you to take another look whenever you are free 😄

**Review by frankieroberto — CHANGES_REQUESTED** [review]
Bug is fixed, thanks @avinhurry!

The only other minor bug I can now spot is that if you go from the review page to the "Right to work or study in the UK" page, the back link takes you to the nationality page - instead it should go back to the review page. Back link works fine on the other pages though. 👍

**Comment by frankieroberto** [comment]
One other thing to check (but I'm not sure exactly how): can we check how this appears on Manage, and in the API? If the user selects one of the radio options (eg "EU Settled Status") then that should appear in Manage, but if they select "Other" then the contents of the text box should appear in Manage (rather than "Other").  Same with the API.

**Review by stevehook — APPROVED** [review]
Looks good to me!

**Pull request merged**